src/LaunchDarkly.ServerSdk/LaunchDarkly.ServerSdk.csproj

```diff
@@ -25,6 +25,7 @@
     <PackageReference Include="LaunchDarkly.Cache" Version="1.0.2" />
     <PackageReference Include="Newtonsoft.Json" Version="9.0.1" Condition="'$(TargetFramework)' != 'net45'" />
     <PackageReference Include="Newtonsoft.Json" Version="6.0.1" Condition="'$(TargetFramework)' == 'net45'" />
+    <PackageReference Include="System.Collections.Immutable" Version="1.5.0" />

```

Comment:
Same package already used by `LaunchDarkly.CommonSdk`:

https://github.com/launchdarkly/dotnet-sdk-common/blob/master/src/LaunchDarkly.CommonSdk/LaunchDarkly.CommonSdk.csproj#L18

Reply:
You're looking at the wrong version of `CommonSdk`, though. What's on master is a later major version, which is used by the Xamarin SDK. The .NET SDK (as you can see in this very same file) is still using version 2.x, which does not have a `System.Collections.Immutable` dependency.

That's what I was getting at earlier when I said we were holding back so as not to add a new dependency. It's not that it has a huge footprint, but whenever we do that, there's a risk that we will cause problems for existing apps that already had a dependency on that package but did not have a binding redirect (because they didn't need one). So, we added it in Xamarin SDK 1.0 where there was no existing code to break, and we can add it in the next major version of .NET. But I'm not sure we can do it right now.

Reply:
Cool.  I'll leave the PR open, and move forward with it in our fork.

Reply:
Well, hold on - maybe my concern is misguided. The third-party dependency where we've had version conflict issues in the past is `Newtonsoft.Json`, but that one has a lot of major versions with widely varying framework compatibility. `System.Collections.Immutable` has changed very little in terms of the core API, is already compatible with every framework we support, and is still on 1.x. And the default behavior without a binding redirect is that the application's dependency version takes priority over transitive dependencies... so the worst that can happen is that the app has a newer version and we end up using that, which shouldn't be a problem. Does that make sense?

Reply:
Sorry, didn't mean that as a bad thing.  We actually maintain a fork because of `Newtonsoft.Json`.    That library is an absolute disaster as far as version compatibility.

I agree that `System.Collections.Immutable` is far more stable and better maintained.  Not sure what your plans are with `LaunchDarkly.CommonSdk` or when this library will update to use the latest version, but I feel as though the  `System.Collections.Immutable` dependency is on its way.  You could select an earlier version of `System.Collections.Immutable` if you are worried about compatibility and it contains what you need, but seems as though `LaunchDarkly.CommonSdk` has already declared 1.5.0, and I think it would be less confusing to just roll with that.

Reply:
> We actually maintain a fork because of Newtonsoft.Json. That library is an absolute disaster as far as version compatibility.

No kidding. FYI, our goal in a future major version is to remove the Newtonsoft types from all of our public APIs (using `LdValue` instead) and at that point we should be able to either migrate to something else internally for parsing, or statically link the package (ILMerge) so it's not a NuGet dependency. That's also the point at which the .NET SDK will start using the most recent version of CommonSdk rather than the 2.x versions.

Anyway, after thinking it over some more I think we are going to go ahead and add the S.C.Imm. dependency. Thanks for the nudge.

Reply:
Most of this library should be immutable, so I think it makes sense.